clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -47,22 +47,21 @@
 public class MockProducerTest {
 
     private final String topic = "topic";
-    private final MockProducer<byte[], byte[]> producer = new MockProducer<>(true, new MockSerializer(), new MockSerializer());
     private final ProducerRecord<byte[], byte[]> record1 = new ProducerRecord<>(topic, "key1".getBytes(), "value1".getBytes());
     private final ProducerRecord<byte[], byte[]> record2 = new ProducerRecord<>(topic, "key2".getBytes(), "value2".getBytes());
-
-
+    
     @Test
-    @SuppressWarnings("unchecked")
     public void testAutoCompleteMock() throws Exception {
-        Future<RecordMetadata> metadata = producer.send(record1);
-        assertTrue("Send should be immediately complete", metadata.isDone());
-        assertFalse("Send should be successful", isError(metadata));
-        assertEquals("Offset should be 0", 0L, metadata.get().offset());
-        assertEquals(topic, metadata.get().topic());
-        assertEquals("We should have the record in our history", singletonList(record1), producer.history());
-        producer.clear();
-        assertEquals("Clear should erase our history", 0, producer.history().size());
+        try (MockProducer<byte[], byte[]> producer = new MockProducer<>(true, new MockSerializer(), new MockSerializer())) {
```

Comment:
Yup, this works too as another way of bringing consistency into these test. Sorry for misunderstanding your suggestion earlier. I'll submit an update shortly.